docs/features/tuples.md

```diff
@@ -0,0 +1,227 @@
+
+```C#
+class Base 
+{
+	virtual void foo(ValueTuple<int, int> arg){...}
+}
+class Derived : Base
+{
+	override void foo((int c, int d) arg){...} // valid override, signatures are equivalent
+}
+class Derived2 : Derived 
+{
+	override void foo((int c1, int c) arg){...} // also valid, warning on possible misuse of name 'c' 
+}
+
+class InvalidOverloading 
+{
+	virtual void foo((int c, int d) arg){...}
+	virtual void foo((int x, int y) arg){...}			// invalid overload, signatures are eqivalent
+	virtual void foo(ValueTuple<int, int> arg){...}		// also invalid
+}
+
+```
+Name erasure at runtime 
+--------------
+Importantly, the tuple field names aren't part of the runtime representation of tuples, but are tracked only by the compiler. 
+
+As a result, the field names will not be available to a 3rd party observer of a tuple instance - such as reflection or dynamic code.
+
+In alignment with the identity conversions, a boxed tuple does not retain the names of the fields and will unbox to any tuple type that has the same element types in the same order.
+
+```C#
+object o = (a: 1, b: 2);    		 // boxing conversion 
+var t = ((int foo, int bar))o;		 // unboxing conversion
+```
+
+Target typing
+--------------
+
+A tuple literal is "target typed" when used in a context specifying a tuple type. What that means is that the tuple literal has a "conversion from expression" to any tuple type, as long as the element expressions of the tuple literal have an implicit conversion to the element types of the tuple type.
+
+``` c#
+(string name, byte age) t = (null, 5); // Ok: the expressions null and 5 convert to string and byte
+```
+
+In cases where the tuple literal is not part of a conversion, a tuple is used by its "natural type", which means a tuple type where the element types are the types of the constituent expressions. Since not all expressions have types, not all tuple literals have a natural type either:
+
+``` c#
+var t = ("John", 5);  						//   Ok: the type of t is (string, int)
+var t = (null, 5);    						//   Error: tuple expression doesn't have a type because null does not have a type
+((1,2, null), 5).ToString();    	    	//   Error: tuple expression doesn't have a type
+
+ImmutableArray.Create((()=>1, 1));        	//   Error: tuple expression doesn't have a type because lambda does not have a type
+ImmutableArray.Create((Func<int>)()=>1, 1); //   ok
+```
+
+A tuple literal may include names, in which case they become part of the natural type:
+
+``` c#
+var t = (name: "John", age: 5); // The type of t is (string name, int age)
+t.age++;						// t has field named age.
+```
+
+A successful conversion from tuple expression to tuple type is classified as _ImplictTuple_ conversion, unless tuple's natural type matches the target type exactly, in such case it is an _Identity_ conversion.
```

Comment:
The names are excluded. I need to mention that. 
It is not very important though, since there cannot be overloads differing only in field names.